### PR TITLE
Add note about the need to fully quality event names in schedule statement docs

### PR DESCRIPTION
**Pull request by ckreibich**
As suggested by @anthonykasza — what do you think Anthony?

### CODE REVIEW

**Comment by anthonykasza** [comment]
TLDR
-----
- DO prepend the event namespace within a schedule statement when that event is declared in any namespace besides GLOBAL.
- DO NOT prepend the event namespace within a schedule statement when that event is declared in the GLOBAL namespace (either explicitly or implicitly).


The details
-----------
Specifying the full namespace only behaves as expected when the namespace is not GLOBAL.  The following 10 scripts demonstrate the above 2 rules. Script four, five, and six show what happens when including GLOBAL in the event's name in the schedule statement.

|script name    | from event            | schedule statement namespace | event declaration namespace    | behavior |
|---------------|-----------------------|------------------------------|---------------------------------------------|------------------------------|
| one.zeek           |`zeek_init()`          | implicit GLOBAL               | implicit GLOBAL               | expected |
| two.zeek           |`zeek_init()`          | explicit GLOBAL               | implicit GLOBAL               | expected |
| three.zeek         |`zeek_init()`          | explicit Foo                  | implicit GLOBAL               | expected |
| four.zeek          |`zeek_init()`          | implicit GLOBAL               | explicit GLOBAL               | unexpected   |
| five.zeek          |`zeek_init()`          | explicit GLOBAL               | explicit GLOBAL               | unexpected  |
| six.zeek           |`zeek_init()`          | explicit Foo                  | explicit GLOBAL               | unexpected   |
| seven.zeek         |`zeek_init()`          | implicit GLOBAL               | explicit Foo                  | expected |
| eight.zeek         | `zeek_init()`         | explicit GLOBAL               | explicit Foo                  | expected |
| nine.zeek          | `zeek_init()`         | explicit Foo                  | explicit Foo                  | expected |

one.zeek
```
event e1() { print "e1"; }

event zeek_init() {
  schedule 1sec { e1() };
}
```


two.zeek
```
event e1() { print "e1"; }

module GLOBAL;

event zeek_init() {
  schedule 1sec { e1() };
}
```

three.zeek
```
event e1() { print "e1"; }

module Foo;

event zeek_init() {
  schedule 1sec { e1() };
}
```

four.zeek
```
event e1() { print "e1"; }

event zeek_init() {
  schedule 1sec { GLOBAL::e1() };
}
```

five.zeek
```
module GLOBAL;
event e1() { print "e1"; }

event zeek_init() {
  schedule 1sec { GLOBAL::e1() };
}
```

six.zeek
```
module GLOBAL;
export {
  global e1: event();
}
event e1() { print "e1"; }

module Foo;
event zeek_init() {
  schedule 1sec { GLOBAL::e1() };
}
```

seven.zeek
```
event zeek_init() {
  # Schedule the event before it's declared isn't allowed.
  schedule 1sec { Foo::e1() };
}

module Foo;
export {
  global e1: event();
}
event e1() { print "e1"; }
```

eight.zeek
```
module Foo;
export {
  global e1: event();
}
event e1() { print "e1"; }

module GLOBAL;
event zeek_init() {
  # Compare to seven.zeek
  # schedule the event after it's been declared is a-ok.
  schedule 1sec { Foo::e1() };
}
```

eight-and-a-half.zeek
```
module Foo;
export {
  global e1: event();
}

module GLOBAL;
event zeek_init() {
  # Compare to seven.zeek and eight.zeek
  # schedule the event after it's been declared is a-ok....
  schedule 1sec { Foo::e1() };
}

module Foo;
# ... even if the event isn't defined until after the schedule statement
event e1() { print "e1"; }
```

nine.zeek
```
module Foo;
export {
  global e1: event();
}
event e1() { print "e1"; }

event zeek_init() {
  schedule 1sec { Foo::e1() };
}
```

**Comment by ckreibich** [comment]
I see — I overlooked the focus on `GLOBAL`. I've included it in the existing longer explanation. I wouldn't include all the examples, it seems overwhelming. Do you think it's necessary?

**Comment by anthonykasza** [comment]
No need to include the examples. You captured the gist well. Thank you!

**Comment by ckreibich** [comment]
Cool, thanks Anthony.
